lib/service/invoices.go

```diff
@@ -159,35 +158,29 @@ func (svc *LndhubService) PayInvoice(ctx context.Context, invoice *models.Invoic
 		Amount:          invoice.Amount,
 	}
 
-	// Start a DB transaction
-	// We rollback anything on error (only the invoice that was passed in to the PayInvoice calls stays in the DB)
-	tx, err := svc.DB.BeginTx(ctx, &sql.TxOptions{})
-	if err != nil {
-		return nil, err
-	}
-
 	// The DB constraints make sure the user actually has enough balance for the transaction
 	// If the user does not have enough balance this call fails
-	_, err = tx.NewInsert().Model(&entry).Exec(ctx)
+	_, err = svc.DB.NewInsert().Model(&entry).Exec(ctx)
 	if err != nil {
-		tx.Rollback()
 		return nil, err
 	}
 
 	// TODO: maybe save errors on the invoice?
 
 	var paymentResponse SendPaymentResponse
 	// Check the destination pubkey if it is an internal invoice and going to our node
+	// Here we start using context.Background because we want to complete these calls
+	// regardless of if the request's context is canceled or not.
 	if svc.IdentityPubkey == invoice.DestinationPubkeyHex {
-		paymentResponse, err = svc.SendInternalPayment(ctx, &tx, invoice)
+		paymentResponse, err = svc.SendInternalPayment(context.Background(), invoice)
 		if err != nil {
-			tx.Rollback()
+			svc.HandleFailedPayment(context.Background(), invoice, entry, err)
 			return nil, err
 		}
 	} else {
-		paymentResponse, err = svc.SendPaymentSync(ctx, &tx, invoice)
+		paymentResponse, err = svc.SendPaymentSync(context.Background(), invoice)
```

Comment:
here context.background was missing, also on handleSuccessfulPayment, added it on both places
does it make sense to use same context for couple of these or new one for all?

Reply:
I think it doesn't really matter, we can use a new one to make it more clear. It's an empty context anyway.